src/providers/virtualraster/qgsvirtualrasterprovider.h

```diff
@@ -0,0 +1,95 @@
+/***************************************************************************
+   qgsvirtualrasterprovider.h - QgsRasterDataProvider
+     --------------------------------------
+    Date                 : June 10, 2021
+    Copyright            : (C) 2021 by Francesco Bursi
+    email                : francesco.bursi
+***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSVIRTUALRASTERPROVIDER_H
+#define QGSVIRTUALRASTERPROVIDER_H
+
+#include <QObject>
+#include "qgsrasterdataprovider.h"
+#include "qgsprovidermetadata.h"
+#include "qgsrastercalculator.h"
+
+class QgsVirtualRasterProvider : public QgsRasterDataProvider
```

Comment:
would be nice to have some more extended doc here for developers. what is virtual raster provider, how it is used etc..